### PR TITLE
mgr/dashboard: fix alert broken for multiple alerts

### PR DESCRIPTION
**Pull request by nizamial09**
After carbon the alert panel was broken when there are multiple alerts present (telemetry, motd, password expiration).

Screenshot from 2024-06-19 11-51-15

Fixes: https://tracker.ceph.com/issues/66512

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by ivoalmeida — APPROVED** [review]
lgtm

**Comment by afreen23** [comment]
I think the banner notif would fit here more
image
See this https://carbondesignsystem.com/patterns/notification-pattern/#banner

**Review by afreen23 — CHANGES_REQUESTED** [review]
I think the banner notif would fit here more
image
See this https://carbondesignsystem.com/patterns/notification-pattern/#banner

**Comment by nizamial09** [comment]
thanks @afreen23 I've applied the banner here. Carbon doesn't natively provide the banner though even though its in their guidelines. So I had to make some extra tweaks inorder to bring the banner.

Screenshot from 2024-06-19 11-22-22

**Review by afreen23 — APPROVED** [review]
looks nice the UX changes, thanks @nizamial09 for beautifying 🎉

**Comment by nizamial09** [comment]
jenkins test api

File: src/pybind/mgr/dashboard/frontend/src/app/shared/components/alert-panel/alert-panel.component.ts [hunk 1]
```diff
@@ -36,7 +50,19 @@ export class AlertPanelComponent implements OnInit {
 
   icons = Icons;
 
+  notificationContent: NotificationContent;
+
   ngOnInit() {
+    const type = this.type === 'danger' ? 'error' : this.type;
```
[the remark this answers is not in the document]
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/components/alert-panel/alert-panel.component.ts]: thank you. I used the NotificationType enum from the carbon-components and now its perfect

File: src/pybind/mgr/dashboard/frontend/src/app/shared/components/telemetry-notification/telemetry-notification.component.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <cd-alert-panel *ngIf="displayNotification"
-                class="no-margin-bottom telemetry-notification"
+                class="no-margin-bottom"
```
[the remark this answers is not in the document]
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/components/telemetry-notification/telemetry-notification.component.html:2]: agreed. we can remove it separately.

**Comment by nizamial09** [comment]
jenkins test make check

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Pull request merged**